components/templates/CodingLayout.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import { allLessons, Lesson } from 'contentlayer/generated'
+import { useState, useEffect } from 'react'
+import { useMediaQuery } from 'react-responsive'
+import { Tabs } from 'components/ui/Tabs'
+import { Info } from './Info'
+import { Code } from './Code'
+
+/**
+ * @slug {string} for fetching challenge data
+ * @solutionScriptSigHex {string} answer to the challenge problem
+ * @next {string} link to next part of chapter
+ */
+export default function CodingLayout({ 
```

Comment:
I don't think `CodingLayout` is the right name for this. More like `TerminalLessonLayout` or so. Later lessons will involve actual scripting, so `CodingLayout` might be more appropriate for those.